assets/src/blocks/HubspotForm/hooks/useHubspotForm.js

```diff
@@ -0,0 +1,66 @@
+import { useState, useEffect } from '@wordpress/element';
+
+export const useHubspotForm = (shortcode, targetRef) => {
+  const [ submitted, setSubmitted ] = useState(false);
+  const [ formId, setFormId ] = useState();
+  const [ portalId, setPortalId ] = useState();
+  const [ submittedMessage, setSubmittedMessage ] = useState();
+
+  const createHubspotForm = () => {
+    window.hbspt.forms.create({
+      portalId,
+      formId,
+      target: `#${targetRef.current.attributes.id.value}`,
+      region: "",
+      onFormSubmitted: ($form) => {
+        setSubmitted(true);
+        if($form.length) {
+          setSubmittedMessage($form[0].innerText);
+        }
+      }
+    });
+  }
+
+  const fetchHubspotFormsLibrary = () => {
```

Comment:
@Inwerpsel unfortunately your suggestion is not possible since this method is calling another function `createHubspotForm` which is also inside the hook and is on charge of modifying `submitted` and `submittedMessage` hook state variables and this is why I added inside the hook.